src/services/storageService.ts

```diff
@@ -27,29 +27,84 @@ export const getPin = async (): Promise<string> => {
 
 export const storeUser = async ({ user }: IStoreUserArgs): Promise<void> => {
   debug(`storeUser(${JSON.stringify(user)})...`)
-  const users_value = await AsyncStorage.getItem(STORAGE_USERS_KEY)
-  if (users_value === null) {
-    const users = new Map<string, IUser>()
+
+  AsyncStorage.getItem(STORAGE_USERS_KEY)
+  .then((result: string | null) => {
+    if (!result) {
+      debug(`storeUser(${JSON.stringify(user)}) no users found`)
+      const users = new Map<string, IUser>()
+      users.set(user.id, user)
+
+      return AsyncStorage.setItem(STORAGE_USERS_KEY, JSON.stringify(Array.from(users)))
+    }
+
+    debug(`storeUser(${JSON.stringify(user)}) users found`)
+    const users: Map<string, IUser> = new Map<string, IUser>();
+    JSON.parse(result).forEach(([key, value]: [string, IUser]) => {
+      users.set(key, value);
+    })
+
+    if (users.get(user.id)) {
+      return Promise.reject('User already exists')
+    }
+
     users.set(user.id, user)
 
-    return AsyncStorage.setItem(STORAGE_USERS_KEY, JSON.stringify(users))
-  }
+    debug(`storeUser(${JSON.stringify(user)}) storing user`)
+    AsyncStorage.setItem(STORAGE_USERS_KEY, JSON.stringify(Array.from(users)))
+  })
+  .catch((error: Error) =>
+    Promise.reject(new Error(`Unable to set value for key: ${STORAGE_PIN_KEY}. ${error.message}`))
+  )
 
-  const users: Map<string, IUser> = JSON.parse(users_value)
-  users.set(user.id, user)
 
-  return AsyncStorage.setItem(STORAGE_USERS_KEY, JSON.stringify(users))
+  // const result: string | null = await AsyncStorage.getItem(STORAGE_USERS_KEY)
```

Comment:
if we don't need these code block, it's better to delete them, and if we do need them at a later point, please make a comment

Reply:
i think i made the commit when you wrote your comment :P